### ISSUE TITLE
LAC Torch version is slower than TF1

### ISSUE THREAD

**Issue by rickstaa** [rickstaa/LAC-TF2-TORCH-translation#18]
**Describe the bug**
When we run the same experiment in torch and TF1 the torch version takes significantly longer.

**To Reproduce**
Steps to reproduce the behaviour:
1. Run the `train.py` file in the `LAC_CLEANED_TORCH` folder.
2. Run the `train.py` file in the `LAC_TF1_ORIGINAL` folder.

**Expected behaviour**
Similar speeds.

**Screenshots**
For training 4 rollouts we get the following results. Left is torch right TensorFlow:

![image](https://user-images.githubusercontent.com/17570430/93987909-6af45b00-fd88-11ea-9ae1-e6ff0dea8e3b.png)

**Desktop (please complete the following information):**
 - OS: Ubuntu 18.04

**Possible debug steps**
- [x] Check whether each operation is efficient.
- [x] Check if gradients are detached if not used anymore.
- [x] Check floating point precission.
- [x] Check if operations that are faster on the GPU are placed on the GPU.
- [x] Use the [torch](https://towardsdatascience.com/efficient-pytorch-part-1-fe40ed5db76c) and [python](https://julien.danjou.info/guide-to-python-profiling-cprofile-concrete-case-carbonara/) profilers to see where the bottlenecks are.
- [x] rickstaa/LAC_TF2_TORCH_REWRITE#17 Check why tf2 is slower than tf1.
